scripts/perf_test.sh

```diff
@@ -0,0 +1,50 @@
+#!/usr/bin/env bash
+set -x
+
+CURRENT_HEAD=$(git rev-parse --abbrev-ref HEAD)
+# If the current state is detached head (e.g. travis), store current commit info instead.
+if [ ${CURRENT_HEAD} = 'HEAD' ]; then
+    git checkout -b tmp
+    CURRENT_HEAD=$(git rev-parse HEAD)
+fi
+# Reference is with respect to the `dev` branch.
+REF_HEAD=${1:-dev}
+BENCHMARK_FILE=tests/perf/test_benchmark.py
+IS_BENCHMARK_FILE_IN_DEV=1
+REF_EXIT_CODE=0
+IS_DIRTY=0
+
+# If uncommitted changes exist, stash these and set a flag.
+git diff-index --quiet HEAD -- || IS_DIRTY=1
+git stash
```

Comment:
Yes, we could definitely make this safer for local runs by skipping the first step if `CI not in os.environ`, and expect the user to generate the profiles on dev. Let me explore a few options here. While I have tested it under several conditions, it can definitely cause weird side effects, for instance if you are running tests on your current branch in a separate terminal.